File: 0260-single-number-iii/0260-single-number-iii.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        diff = reduce(lambda a, b: a ^ b, nums)
        diff &= -diff
        
        val1 = val2 = 0
        for num in nums:
            if(num & diff > 0):
                val1 ^= num
            else:
                val2 ^= num
        
        return [val1, val2]

File: 0260-single-number-iii/0260-single-number-iii.go
func singleNumber(nums []int) []int {
    var xored = 0
    for _, num := range nums {
        xored ^= num
    }
    
    xored &= -xored
    var val1, val2 = 0, 0
    for _, num := range nums {
        if xored & num > 0{
            val1 ^= num
        } else {
            val2 ^= num
        }
    }
    
    return []int{val1, val2}
}